website/src/app/blog/how-dns-works-in-firezone/readme.mdx

```diff
@@ -1,18 +1,383 @@
-# How Firezone DNS Works
+import Image from "next/image";
+import Alert from "@/components/DocsAlert";
+import Link from "next/link";
 
-_Note: This blog post describes in detail how Firezone DNS works in all its
-complexity. If you're looking for a more high-level reference, read the
-[docs page on the subject instead](/kb/deploy/dns)._
+Firezone's approach to DNS works a bit differently than one might expect. One
+question we get a lot is, "why do my DNS Resources resolve to different IPs with
+Firezone enabled?". Great question. Let's explain that now!
+
+What follows is a brief history of DNS, (a few of) the security issues that
+plague it, and how Firezone's DNS-based traffic routing was designed to address
+them.
+
+## A brief history lesson
+
+But before we dive into all the fun details, it's helpful to first understand
+the problem DNS was originally designed to solve.
+
+DNS was created in the early 1980s to address a classic scalability problem with
+the early ARPANET.
+
+<Image
+  src="/images/blog/how-dns-based-traffic-routing-works/arpanet-logical-map.png"
+  alt="Early ARPANET logical map"
+  width={1200}
+  height={1200}
+  className="mx-auto rounded shadow"
+/>
+
+<p className="text-sm italic text-center mx-auto">
+  Figure 1: Early ARPANET logical map
+</p>
+
+The ARPANET, the precursor to the modern internet, was a network of computers
+that connected various research institutions and government agencies across the
+United States. It was the first network to use the new TCP/IP protocol suite,
+which provided an IP address to each host on the network.
+
+To connect to a host on the ARPANET, you needed its IP address. But there was a
+problem: IP addresses are hard to remember, and it turns out humans aren't great
+at remembering long strings of numbers. To solve this, a simple system was
+created to map each host's IP address to a human-friendly "hostname".
+
+Simple enough, right? Well, not quite. The system was perhaps a bit _too_
+simple.
+
+### Files don't scale
+
+Hostname to IP address mappings were stored in a single file, aptly named
+`HOSTS.TXT`, and stored on the `SRI-NIC` server -- a system maintained by the
+Network Information Center (NIC) at the Stanford Research Institute (SRI).
+
+<p className="text-sm italic text-center mx-auto">
+  Figure 2: Artist's rendition of the original HOSTS.TXT file
+</p>
```

Comment:
Was the image intentionally not put in for now or should this be removed?

Reply:
Still working on the graphics